src/aio/aio_task_queue.h

```diff
@@ -0,0 +1,170 @@
+        }
+    }
+    error_code get_error() const { return _err; }
+    aio_task *get_task() const { return _tsk; }
+    uint32_t get_processed_bytes() const { return _processed_bytes; }
+
+private:
+    std::unique_ptr<utils::notify_event> _notify;
+    aio_task *_tsk;
+    aio_provider *_provider;
+
+    error_code _err;
+    uint32_t _processed_bytes;
+};
+
+class io_event_loop_t
+{
+public:
+    explicit io_event_loop_t(service_node *node)
+    {
+        _is_running = true;
+
+        _thread = std::thread([this, node]() {
+            task::set_tls_dsn_context(node, nullptr);
+
+            std::shared_ptr<io_event_t> evt;
+            int result = 0;
+            while (true) {
+                if (dsn_unlikely(!_is_running.load(std::memory_order_relaxed))) {
+                    break;
+                }
+                if (!_evt_que.wait_dequeue_timed(evt, 200)) {
+                    continue;
+                }
+                evt->complete();
+            }
+        });
+    }
+
+    ~io_event_loop_t()
+    {
+        if (!_is_running) {
+            return;
+        }
+        _is_running = false;
+        _thread.join();
+    }
+
+    void enqueue(std::shared_ptr<io_event_t> evt) { _evt_que.enqueue(std::move(evt)); }
+
+private:
+    // one queue per thread
+    moodycamel::BlockingConcurrentQueue<std::shared_ptr<io_event_t>> _evt_que;
+    std::thread _thread;
+    std::atomic<bool> _is_running{false};
+};
+
+class aio_task_queue
+{
+public:
+    aio_task_queue(disk_engine *disk)
+    {
+        for (int i = 0; i < 8; i++) {
+            _high_pri_workers.push_back(std::make_shared<io_event_loop_t>(disk->node()));
+        }
+        for (int i = 0; i < 4; i++) {
+            _comm_pri_workers.push_back(std::make_shared<io_event_loop_t>(disk->node()));
+        }
+        for (int i = 0; i < 2; i++) {
+            _low_pri_workers.push_back(std::make_shared<io_event_loop_t>(disk->node()));
+        }
+    };
+    ~aio_task_queue(){};
+
+    void enqueue(std::shared_ptr<io_event_t> evt)
+    {
+        dsn_task_priority_t priority = task_spec::get(evt->get_task()->code())->priority;
+        int task_hash = evt->get_task()->hash();
+        switch (priority) {
+        case dsn_task_priority_t::TASK_PRIORITY_HIGH:
+            _high_pri_workers[task_hash % _high_pri_workers.size()]->enqueue(evt);
+            break;
+        case dsn_task_priority_t::TASK_PRIORITY_COMMON:
+            _comm_pri_workers[task_hash % _comm_pri_workers.size()]->enqueue(evt);
+            break;
+        case dsn_task_priority_t::TASK_PRIORITY_LOW:
+            _low_pri_workers[task_hash % _low_pri_workers.size()]->enqueue(evt);
+            break;
+        default:
+            dassert_f(false,
+                      "task priority must be set "
+                      "TASK_PRIORITY_HIGH/TASK_PRIORITY_COMMON/TASK_PRIORITY_LOW");
+            break;
+        }
+    };
+
+private:
+    std::vector<std::shared_ptr<io_event_loop_t>> _high_pri_workers;
```

Comment:
`map<dsn_task_priority_t, std::vector<std::shared_ptr<io_event_loop_t>>>`
so the func `enqueue` can be more simplify